server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -774,7 +797,7 @@ public synchronized ObjectLongMap<String> getInSyncGlobalCheckpoints() {
         final ObjectLongMap<String> globalCheckpoints = new ObjectLongHashMap<>(checkpoints.size()); // upper bound on the size
         checkpoints.entrySet()
             .stream()
-            .filter(e -> e.getValue().inSync)
+            .filter(e -> e.getValue().inSync && e.getValue().localTranslog)
```

Comment:
What would be the guideline to use `localTranslog` field? Should it be used each time when we are checking if `inSync` is true or not?

Reply:
`localTranslog` signifies whether for the shard the translog store is local FS or not. This is used for replication semantics. for eg - computation of global checkpoint, fanning out to replication requests, need for retention leases to name couple of them.